cli/lib/nf3.mjs

```diff
@@ -111,6 +116,29 @@ class Nf3 {
     this.ethereumSigningKey = ethereumSigningKey;
     this.zkpKeys = zkpKeys;
     this.currentEnvironment = environment;
+    this.clientAuthenticationKey = clientApiAuthenticationKey;
+
+    this.applyHttpClientAuthentication();
+  }
+
+  applyHttpClientAuthentication() {
+    if (!this.clientAuthenticationKey) {
+      logger.info('No client authentication key is set!');
```

Comment:
Don't we want to throw at this point, and use `console.error` at this point?

Reply:
No, because this feature is not required to be setup. It will depend on the deployer's wish.